File: Semestru 4/MPP/lab/.net/lab3/lab3/Server/service/ServiceEvent.cs
using Model.app.domain;
using Persistence.app.repo.@interface;
using Services.services;

namespace Server.app.service
{
	public class ServiceEvent : IServiceEvent
	{
		private IEventRepository Repo;

		public ServiceEvent(IEventRepository repo) =>
			this.Repo = repo;

		public IEnumerable<Event> GetAll() =>
			this.Repo.GetAll();

		public Event? GetById(int id) =>
			this.Repo.GetById(id);

		public Event? Update(Event event_) =>
			this.Repo.Update(event_);

		public IEnumerable<Event>? GetByMinAge(int minAge) =>
			this.Repo.GetAllByMinAge(minAge);

		public IEnumerable<Event>? GetByMaxAge(int maxAge) =>
			this.Repo.GetAllByMaxAge(maxAge);

		public Event? Add(string name, int minAge, int maxAge) => 
			this.Repo.Create(new Event(name, minAge, maxAge));
	}
}

File: Semestru 4/MPP/lab/java/lab3/Services/src/main/java/org/example/services/ServiceEvent.java
package org.example.services;

import org.example.domain.Event;
import org.example.repository.interf.IEventRepository;

public class ServiceEvent implements IServiceEvent {
    private IEventRepository Repo;

    public ServiceEvent(IEventRepository repo) {
        this.Repo = repo;
    }

    public Iterable<Event> GetAll() {
        return this.Repo.FindAll();
    }

    public Event GetById(int id) {
        return this.Repo.FindById(id);
    }

    public Event Add(String name, int minAge, int maxAge) {
        return this.Repo.Save(new Event(name, minAge, maxAge));
    }

    public Event UpdateEvent(Event event) {
        return this.Repo.Update(event);
    }
}
